src/blocked.c

```diff
@@ -98,7 +98,9 @@ void freeClientBlockingState(client *c) {
     dictRelease(c->bstate->keys);
     zfree(c->bstate);
     c->bstate = NULL;
-    if (c->deferred_reply) commitDeferredReplyBuffer(c);
+
+    serverAssert(!c->deferred_reply);
```

Comment:
Use debugServerAssert makes it crash only during our tests and not in production. It is enabled by the undocumented config `enable-debug-assert` which is off by default.

```suggestion
    debugServerAssert(!c->deferred_reply);
```